apps/storefront/src/pages/Dashboard/index.mobile.test.tsx

```diff
@@ -0,0 +1,446 @@
+
+    await waitFor(() => {
+      expect(screen.queryByText(/Loading/)).not.toBeInTheDocument();
+    });
+
+    const monstersInc = buildCompanyEdgeWith({ node: { companyName: 'Monsters Inc' } });
+
+    getCompanies.mockReturnValue(
+      buildCompanyListWith({
+        data: { superAdminCompanies: { edges: [monstersInc], totalCount: 1 } },
+      }),
+    );
+
+    const searchBox = screen.getByPlaceholderText(/Search/);
+
+    await userEvent.type(searchBox, 'monsters');
+
+    await waitFor(() => {
+      expect(screen.getByText(/Monsters Inc/)).toBeInTheDocument();
+    });
+
+    expect(getCompanies).toHaveBeenLastCalledWith(expect.stringContaining('search: "monsters"'));
+  });
+
+  it('allows user to change the number of rows per page', async () => {
+    const getCompanies = vi.fn().mockReturnValue(
+      buildCompanyListWith({
+        data: {
+          superAdminCompanies: {
+            edges: bulk(buildCompanyEdgeWith, 'WHATEVER_VALUES').times(10),
+          },
+        },
+      }),
+    );
+
+    server.use(
+      graphql.query('SuperAdminCompanies', ({ query }) => HttpResponse.json(getCompanies(query))),
+    );
+
+    renderWithProviders(<Dashboard setOpenPage={vi.fn()} />, { preloadedState });
+
+    await waitFor(() => {
+      expect(screen.queryByText(/Loading/)).not.toBeInTheDocument();
+    });
+
+    getCompanies.mockReturnValue(
+      buildCompanyListWith({
+        data: {
+          superAdminCompanies: {
+            edges: bulk(buildCompanyEdgeWith, 'WHATEVER_VALUES').times(20),
+          },
+        },
+      }),
+    );
+
+    const rowsPerPageSelect = screen.getByRole('combobox', { name: /per page:/ });
+
+    await userEvent.click(rowsPerPageSelect);
+
+    const option = screen.getByRole('option', { name: '20' });
+
+    await userEvent.click(option);
+
+    await waitFor(() => {
+      expect(screen.getAllByRole('button', { name: /Masquerade/ })).toHaveLength(20);
+    });
+
+    expect(getCompanies).toHaveBeenLastCalledWith(expect.stringContaining('first: 20'));
+  });
+
+  it('can go to the next page', async () => {
+    const getCompanies = vi.fn().mockReturnValue(
+      buildCompanyListWith({
+        data: {
+          superAdminCompanies: {
+            edges: bulk(buildCompanyEdgeWith, 'WHATEVER_VALUES').times(10),
+            totalCount: 100,
+          },
+        },
+      }),
+    );
+
+    server.use(
+      graphql.query('SuperAdminCompanies', ({ query }) => HttpResponse.json(getCompanies(query))),
+    );
+
+    renderWithProviders(<Dashboard setOpenPage={vi.fn()} />, { preloadedState });
+
+    await waitFor(() => {
+      expect(screen.queryByText(/Loading/)).not.toBeInTheDocument();
+    });
+
+    const monstersInc = buildCompanyEdgeWith({ node: { companyName: 'Monsters Inc' } });
+
+    getCompanies.mockReturnValue(
+      buildCompanyListWith({
+        data: { superAdminCompanies: { edges: [monstersInc], totalCount: 100 } },
+      }),
+    );
```

Comment:
are we missing a user-facing assertion which uses this mock?

Reply:
we are, too many tests, all too similar 🤦